### PR TITLE
Fix data picker initial state

### PR DESCRIPTION
**Pull request by alxnddr**
### Description

The issue existed even before the newly added saved question picker.

#### Steps to reproduce:
- Ask a simple question
- Select any saved question
- After the visualization is rendered press "Show editor"
Screen Shot 2021-07-06 at 00 25 30
- Open the data picker again

Before this PR:
It opened the list of tables from the database is used in the selected saved question

After this PR:
It opens the saved question picker with the question selected

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/saved-question-picker/utils.js [hunk 1]
```diff
@@ -22,3 +22,23 @@ export function buildCollectionTree(collections) {
...
+  }
+
+  return findCollectionByName(
+    collections
+      .map(c => c.children)
+      .filter(Boolean)
+      .flat(),
+    name,
+  );
+};
```
Comment (kulyk) [inline frontend/src/metabase/query_builder/components/saved-question-picker/utils.js:44]: Can it return an unexpected collection if it happened to have the same name?
If so, can we somehow narrow down the search? Maybe we'll need a backend change?

**Review by kulyk — COMMENTED** [review]
Looks and works great 👍

File: frontend/src/metabase/query_builder/components/data-search/utils.js [hunk 2]
```diff
@@ -12,3 +12,10 @@ export function convertSearchResultToTableLikeItem(searchResultItem) {
 
   return searchResultItem;
 }
+
+export function isSavedQuestion(tableId) {
+  return (
+    typeof tableId === "string" &&
+    tableId.indexOf(SAVED_QUESTION_ID_PREFIX) === 0
```
[the remark this answers is not in the document]
Comment (alxnddr) [reply frontend/src/metabase/query_builder/components/data-search/utils.js]: I forgot about `startsWith`, thanks!

**Review by flamber — COMMENTED** [review]
Nice. Have been an annoyance for a long time. Would be great if it could show the selected question instead of have to scroll to find it, but that's minor.
I can see the problem with names vs IDs - it causes problems for colons in names too #14287

File: frontend/src/metabase/query_builder/components/saved-question-picker/utils.js (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/query_builder/components/saved-question-picker/utils.js:44]: This requires some BE change, @dpsutton and I will take a look separately from this PR

**Comment by alxnddr** [comment]
@flamber thanks for pointing out the need to scroll to the items. I'll add it in a follow-up. I already created one for collection names with colons https://github.com/metabase/metabase/pull/16904

**Review by flamber — APPROVED** [review]

**Pull request merged**